docs/source/api/apollo-server.md

```diff
@@ -369,3 +369,19 @@ addMockFunctionsToSchema({
 *  `maskErrorDetails`: boolean
 
    Set to true to remove error details from the traces sent to Apollo's servers. Defaults to false.
+
+*  `generateClientInfo`: (GraphQLRequestContext) => ClientInfo **AS 2.2**
+
+   Creates a client context(ClientInfo) based on the request pipeline's
+   context, which contains values like the request, response, cache, and
+   context. This generated client information will be provided to Apollo
+   Engine and can be used to filter metrics. Set `clientName` to identify a
+   particular client. Use `clientVersion` to specify a version for a client
+   name.  The default function will use the `clientInfo` field inside of
+   GraphQL Query `extensions`.
+
+   For some advanced cases, the `clientReferenceId` can be used as a unique
+   identifier for a client name. These clientReferenceIds should have a one
```

Comment:
This `clientReferenceId`*
Or maybe this would be more clear if it said something like 
```
[For advanced use cases when you already have an opaque string to identify your client (e.g. an API key, x509 certificate, or team codename), use the `clientReferenceId` field to add a reference to your reported client and its internal identity. This client reference ID will not be displayed in the UI but will be available for cross-correspondence. 

>[WARNING] If you specify a `clientReferenceId`, Engine will treat the `clientName` as a secondary lookup, so changing a clientName may result in an unwanted experience.]
```
I do want to say something explicit about the weird re-naming behavior with a big ⚡️ WARNING ⚡️ , but open to ideas about how best to say it.